### PR TITLE
Add `Float`s to `enum Const`.

### PR DESCRIPTION
**Pull request by ltratt**
This makes clear that `Const` can't be `Eq` because floats aren't. We therefore implement a new wrapper that is solely used for storing `Const`s in an `IndexSet`, without giving users the idea that `Const` can be compared for equailty.

Note that this doesn't fully implement all the necessary bits, but it does enough that it should make clear what extra bits need doing.

### CODE REVIEW

**Comment by vext01** [comment]
https://github.com/ykjit/yk/pull/1267/commits/e327e47a04d0d8e124ab32dff3f4fd05ed446f01 implements the rest of what is needed for float consts.

Requires https://github.com/ykjit/ykllvm/pull/177

File: ykrt/src/compile/jitc_yk/jit_ir/mod.rs [hunk 1]
```diff
@@ -953,18 +964,71 @@ impl fmt::Display for DisplayableOperand<'_> {
...
+///
+/// Note that this struct deliberately does not implement `PartialEq` (or `Eq`): two instances of
+/// `Const` may represent the same underlying constant, but (because of floats), you as the user
+/// need to determine what notion of equality you wish to use on a given const.
+#[derive(Clone, Debug)]
 pub(crate) enum Const {
     /// A constant integer at most 64 bits wide. This can be treated a signed or unsigned integer
     /// depending on the operations that use this constant (the [Ty::Integer] type itself has no
     /// concept of signedness).
+    #[allow(dead_code)]
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir/mod.rs:976]: I suppose this `#[dead_code]` annotation can go now.

Comment (vext01) [reply ykrt/src/compile/jitc_yk/jit_ir/mod.rs]: Fixed in https://github.com/ykjit/yk/pull/1267/commits/c96852d70914bf6e35bd7c9ddabf2be19b4a9f30

**Comment by ltratt** [comment]
Lukas is away so we'll have to break our normal rule and I'll be the reviewer for your code, and you can review mine. I'm fine with your code; if so, please squash (but you should probably leave my commit as-is).

**Comment by vext01** [comment]
Squashed.

**Comment by vext01** [comment]
Happy with https://github.com/ykjit/yk/pull/1267/commits/2596fae3d0ba40821531725f220f80680248b64b ?

**Comment by ltratt** [comment]
I'm not happy with it. I'm ecstatic! I didn't know there was a non-unsafe way doing this, and I'm happy you are less ignorant than I am! Please squash.

**Comment by vext01** [comment]
Well, clippy caught it :P I only learned about it today!

**Comment by vext01** [comment]
Squashed, but, hah, partially verified commit.

If that bothers you, you can resign that and force push.

Otherwise ready to merge.

**Comment by ltratt** [comment]
Kinda makes sense TBH.

**Pull request merged**